### ISSUE TITLE
Save and Restore - cannot delete snapshots

### ISSUE THREAD

**Issue by UrbanBobek** [ControlSystemStudio/phoebus#2478]
Hi all,

I am having difficulties with deleting snapshots I create in Save and Restore. 

I created the following tree structure:
![image](https://user-images.githubusercontent.com/43409317/205939939-6e06aa5d-78c6-4585-82a8-bd55fdf9c201.png)

When I try to delete any of the snapshots I receive a pop up error message stating that it failed to delete either a folder or a save set.  What is particularly strange is that if I try to delete for example the `calc` snapshot, I receive an error message that the `'Root folder'` could not be deleted. But when I try to delete for example `log_test`, I receive an error message that the `'sub_folder'` could not be deleted.
![image](https://user-images.githubusercontent.com/43409317/205939833-e51258a7-ad6a-4a22-b629-090eab35fd87.png)

When I try to delete the snapshots I get no error messages in the Phoebus console or from the Save and Restore service. I checked the queries sent by the service to the mysql database when I try to delete one of the snapshots (`calc`):
```
2022-12-06T16:21:54.929756Z         8 Query     select n.*, nc.ancestor from node as n join node_closure as nc on n.id=nc.descendant where n.username is not NULL and nc.ancestor=(select id from node where unique_id='47b5725f-9c8c-41fa-b8d9-75e4260112a8') and nc.depth=1
2022-12-06T16:21:54.931641Z         8 Query     select * from property where node_id=6
2022-12-06T16:21:54.933106Z         8 Query     select * from tag where snapshot_id='230f9677-215f-4049-9714-4added63774a'
2022-12-06T16:21:54.936410Z         8 Query     select * from property where node_id=7
2022-12-06T16:21:54.937339Z         8 Query     select * from tag where snapshot_id='ee3f7877-7b33-4deb-a865-aaba1992c4df'
2022-12-06T16:21:54.939685Z         8 Query     select * from property where node_id=8
2022-12-06T16:21:54.942875Z         8 Query     select * from tag where snapshot_id='5e224edc-4817-4376-bec1-309cb5a4883a'
2022-12-06T16:21:54.943705Z         8 Query     select * from property where node_id=9
2022-12-06T16:21:54.944578Z         8 Query     select * from tag where snapshot_id='689b2ee0-049c-4e1e-8a16-1d9d8e96cd2f'
2022-12-06T16:21:54.949116Z         8 Query     select * from property where node_id=10
2022-12-06T16:21:54.955695Z         8 Query     select * from tag where snapshot_id='91f3a61e-5981-427d-aa86-244494ccb0e4'
```
The database was created using the following commands:
```
CREATE USER 'masar'@'%' IDENTIFIED BY '<password>';
CREATE DATABASE masar;
GRANT ALL PRIVILEGES ON masar.* TO 'masar'@'%';

FLUSH PRIVILEGES;
```
I am not particularly familiar with mysql syntax, but I was able to manually delete some entries from the masar database tables when logged in as the `masar` user.

I tested this with Save and Restore v4.6.6 and v4.7.1. I am aware that this service was recently migrated to ElasticSearch, but unfortunately we are bound to using older versions of these services. 

Any help would be greatly appreciated. 

Thanks!
Urban

**Comment by georgweiss** [comment]
Running code prior to Elasticsearch migration (which corresponds to version 4.7.1), I cannot reproduce. Deleting snapshots or a folder works fine in my setup.

Are there any stack traces on the server side to be found?

**Comment by UrbanBobek** [comment]
I see. Is there a particular version of Mysql after which the service was not compatible anymore? I created the database using mysql versions 8.0.11 and 8.0.4 without success. Since I am using the official mysql docker images to deploy the database I did not try version 8.0.9, because it is not available. 

Thank you for the heads up. The purpose of manually deleting entries in the database was to check if there might be any privilege issues and is not something I would do in a production environment. 

Nothing is printed in the console/Error log when I try to delete a snapshot. Is there anywhere else should check?

**Comment by georgweiss** [comment]
Support for Mysql broke when we updated the Spring Boot version for the project. We did that in order to stay reasonably up-to-date and since we did such an upgrade in other service modules.

Since we were working on a migration to Elasticsearch, I decided to not try to fix the Mysql issue.

Actually an error on the service side should leave something in the log. I realize that otherwise this error leaves too little traces useful for debugging.

I would maybe recommend to run at least Phoebus in an IDE in debug mode. I find IntelliJ (even in community edition) very easy to use with a Maven project like Phoebus.

**Comment by UrbanBobek** [comment]
Did the Spring Boot version upgrade also affect the support for Postgresql? If it didn`t, I assume it would be easier for us migrate from Mysql. 

We would like to deploy the services without the need of upgrading them in the future. Is there an approximate time-frame for when a stable release of the Elasticsearch based services will be available?

I am glad you spotted the bug. Can I ask you for the commit ID once the bug is fixed?

**Comment by georgweiss** [comment]
In my setup Postgresql still works with the updated version of Spring Boot. But I strongly recommend migration to Elasticsearch.

As the migration to Elasticsearch has been merged to the master branch I consider it stable enough to release for production here at ESS. Moreover, I am currently adding features requested by my users.

I will make a separate PR for the bug fix.

**Comment by georgweiss** [comment]
We can consider a release (that would be 4.7.2) in order to make the Elasticsearch migration more "official". But cannot say exactly when that would happen. Probably need to await some more testing/runtime here at ESS.

**Comment by georgweiss** [comment]
@UrbanBobek, the bug fix for the case when the error dialog shows the wrong name is  commit f2c1c1345936e43d86e734bce367bf23fd011354

**Comment by UrbanBobek** [comment]
Of course, it was not my intention to rush you. I am looking forward to the release!

Thank you for the commit id

**Comment by UrbanBobek** [comment]
Upgrading Save and Restore to the current version that uses Elasticsearch resolved our problems, so I am closing the issue.

Again, thanks for the help!

**Comment by georgweiss** [comment]
@UrbanBobek , in case this is still of interest: we released 4.7.2 of the code base which now contains the Elasticsearch-based implementation for save&restore.
